src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -240,6 +240,31 @@ void TlsChannelSecurityConnector::check_peer(
                 : check_arg_->peer_cert_full_chain;
         gpr_free(peer_pem_chain);
       }
+      // Fill in the subject alternative names
+      std::vector<char*> subject_alternative_names;
```

Comment:
I think the logic would be simplified if we can use `std::string` instead of `char*` here. Is it because we have `char**` in `grpc_tls_server_authorization_check_arg`?
My next planned large item is to improve the custom verification. Do you think it would be a good idea if we make `grpc_tls_server_authorization_check_arg` an internal core class instead of a struct exposed in `grpc_security.h`? It now contains many irrelevant fields. 
If we make it an internal class, we will need some helper functions like `get_target_name_from_arg`, `get_peer_cert_from_arg`, `get_san_names_from_arg` in `grpc_security.h` to get the information we really want. But the changes here could be simplified by using `std::string`, since here we don't use the core API. WDYT? You can add a TODO(for cleaning up the code) for me if that makes sense to you.

Reply:
looks like reviewable did not post the response here.

I think the logic would be simplified if we can use std::string instead of char* here. Is it because we have char** in grpc_tls_server_authorization_check_arg?

That is correct. If we can use std::string, the logic would get simpler.

My next planned large item is to improve the custom verification. Do you think it would be a good idea if we make grpc_tls_server_authorization_check_arg an internal core class instead of a struct exposed in grpc_security.h? It now contains many irrelevant fields.

Are you sure we can use an internal Core class from the wrapped C++ layer? What if we want to use this from other languages too?

Added a TODO anyway